modules/waveform-workflowoperation/src/main/java/org/opencastproject/workflow/handler/waveform/WaveformWorkflowOperationHandler.java

```diff
@@ -155,6 +158,10 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
     );
 
     String color = StringUtils.trimToNull(workflowInstance.getCurrentOperation().getConfiguration(COLOR_PROPERTY));
+    if (StringUtils.isBlank(color)) {
```

Comment:
There isn't any real reason the set the default twice, is there?
Plus, we don't always additionally log that we are using the default in info level.

Reply:
> There isn't any real reason the set the default twice, is there?

When I made tests last week I notice that the worker applied the default value at startup, but when it was called the function the `waveFormColor` variable was blank. I think is something with the scope.

Reply:
It's because the remote implementation would send a blank value which then wouldn't meet the criteria of being `null` but `isBlank` will catch that case.

Reply:
Re: setting the default, the WOH's color instance variable is set to null if there is no setting provided in the XML, so yes, we do need to set the default here :)

Reply:
> the WOH's color instance variable is set to `null` if there is no setting provided in the XML

…which should be okay since the service implementation now uses a default value if the incoming color is `null` or blank. If we catch empty values here, the impl wouldn't need a default and could just throw an exception instead to make it clear that it does expect proper values.

Setting the default at two places is confusing and may easily lead to error if we ever want to change the default.

Reply:
Yeah, good point.  Ok, this has been removed.